src/wp-includes/class-wp-theme-json-resolver.php

```diff
@@ -172,8 +172,18 @@ public static function get_core_data() {
 		 *
 		 * @param WP_Theme_JSON_Data $theme_json Class to access and update the underlying data.
 		 */
-		$theme_json   = apply_filters( 'wp_theme_json_data_default', new WP_Theme_JSON_Data( $config, 'default' ) );
-		static::$core = $theme_json->get_theme_json();
+		$theme_json = apply_filters( 'wp_theme_json_data_default', new WP_Theme_JSON_Data( $config, 'default' ) );
+
+		/*
+		 * Backward compatibility for extenders returning a WP_Theme_JSON_Data
+		 * compatible class that is not a WP_Theme_JSON_Data object.
+		 */
+		if ( is_a( $theme_json, 'WP_Theme_JSON_Data' ) ) {
```

Comment:
This should use the `instanceof` keyword, see https://core.trac.wordpress.org/changeset/56352.

Reply:
Thanks! I had a sense that we preferred `instanceof` but couldn't remember the context. Do we not have a sniff for this?

Reply:
Addressed in [544d762](https://github.com/WordPress/wordpress-develop/pull/6626/commits/544d7628992d4bfd53f8176ef6ee1a979b15139a)